stream-video-android-core/src/main/kotlin/io/getstream/video/android/core/call/RtcSession.kt

```diff
@@ -524,7 +542,7 @@ public class RtcSession internal constructor(
      */
     suspend fun reconnect(forceRestart: Boolean) {
         // ice restart
-        val subscriberAsync = coroutineScope.async {
+        val subscriberAsync = rtcSessionScope.async {
```

Comment:
Is `async` truly effective within rtcSessionScope since it's backed by a single thread? If not then maybe we make the code more simple